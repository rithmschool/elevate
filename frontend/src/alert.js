import React, { Component } from "react";

class Alert extends Component {
  static defaultProps = {
    type: "danger",
    messages: []
  };

  render() {
    const { messages, type, link, text } = this.props;
    return (
      <div className={`alert alert-${type}`} role="alert" align="center">
        {messages.map(error => (
          <p className="mb-0 small" key={error}>
            {error}
          </p>
          
        ))}
        <a href={link}>{text}</a>
      </div>
    );
  }
}

export default Alert;
