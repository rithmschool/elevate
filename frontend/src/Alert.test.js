import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Alert from "./Alert";

it("renders without crashing", function() {
  shallow(<Alert />);
});

it("matches snapshot for danger", function() {
  let messages = ["Everything is broken", "Run for the hills"];
  let wrapper = shallow(<Alert type="danger" messages={messages} />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

it("matches snapshot for success", function() {
  let messages = ["Everything is awesome!"];
  let wrapper = shallow(<Alert type="success" messages={messages} />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
