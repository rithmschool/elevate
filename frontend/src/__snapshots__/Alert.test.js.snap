// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for danger 1`] = `
<div
  align="center"
  className="alert alert-danger"
  role="alert"
>
  <p
    className="mb-0 small"
    key="Everything is broken"
  >
    Everything is broken
  </p>
  <p
    className="mb-0 small"
    key="Run for the hills"
  >
    Run for the hills
  </p>
  <a />
</div>
`;

exports[`matches snapshot for success 1`] = `
<div
  align="center"
  className="alert alert-success"
  role="alert"
>
  <p
    className="mb-0 small"
    key="Everything is awesome!"
  >
    Everything is awesome!
  </p>
  <a />
</div>
`;
